metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java

```diff
@@ -46,35 +54,113 @@
+        assertThat(actual, is(equalTo(expectedRequest)));
+    }
+
+    @Test
+    void getCreatePublicationReturnsMostCompletePublicationDateWhenMultipleCandidatesArePresent() throws IOException {
+
+        List<MetaTagPair> metaDates = List.of(new MetaTagPair(DC_DATE, YEAR_ONLY),
+                new MetaTagPair(DC_DATE, FULL_DATE));
+
+        String html = createHtml(metaDates);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
+        MetadataService metadataService = new MetadataService();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+        CreatePublicationRequest actual = request.orElseThrow();
+        actual.setContext(null);
+
+        CreatePublicationRequest expectedRequest = getCreatePublicationRequestWithDateOnly(FULL_DATE);
+        assertThat(actual, is(equalTo(expectedRequest)));
+    }
+
+    @ParameterizedTest(name = "Bad date {0} is ignored")
+    @ValueSource(strings = {"20111-02-01", "2011-033-11", "2010-01-011", "20100101", "First of Sept. 2010"})
+    void getCreatePublicationReturnsValidDateWhenValidAndInvalidCandidatesAreAvailable(String nonsense)
+            throws IOException {
+        List<MetaTagPair> metaDates = List.of(new MetaTagPair(DC_DATE, VALID_DATE),
+                new MetaTagPair(DC_DATE, nonsense));
+
+        String html = createHtml(metaDates);
+        URI uri = prepareWebServerAndReturnUriToMetadata(ARTICLE_HTML, html);
+        MetadataService metadataService = new MetadataService();
+        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);
+        CreatePublicationRequest actual = request.orElseThrow();
+        actual.setContext(null);
+
+        CreatePublicationRequest expectedRequest = getCreatePublicationRequestWithDateOnly(VALID_DATE);
+        assertThat(actual, is(equalTo(expectedRequest)));
+    }
+
+    private CreatePublicationRequest getCreatePublicationRequestWithDateOnly(String date) {
+        EntityDescription entityDescription = new EntityDescription.Builder()
+                .withDate(createPublicationDate(date))
+                .build();
+        CreatePublicationRequest expectedRequest = new CreatePublicationRequest();
+        expectedRequest.setEntityDescription(entityDescription);
+        return expectedRequest;
+    }
+
+    private PublicationDate createPublicationDate(String date) {
+        String[] dateParts = date.split(DATE_SEPARATOR);
+        String year = dateParts[0];
+        String month = dateParts.length > 1 ? dateParts[1] : null;
+        String day = dateParts.length > 2 ? dateParts[2] : null;
+
+        return new PublicationDate.Builder()
+                .withYear(year)
+                .withMonth(month)
+                .withDay(day)
+                .build();
     }
 
     private static Stream<Arguments> provideMetadataWithLowercasePrefixes() {
```

Comment:
and a global @parameterizedTest that takes test names as a parameter...

I feel like we should do just one of these, at most two of them?

Reply:
Will fix this in subsequent PRs